crypto.go

```diff
@@ -469,6 +585,51 @@ func DeriveSigningKey(seed []byte, salt []byte, iter int) (*[32]byte, *[64]byte)
 	return publicKey, privateKey
 }
 
+func DeriveCredentials(username string, password string, realmName string, nameSalt string) (string, EncryptionKeys, SigningKeys, error) {
```

Comment:
function comment, `DeriveIdentityCredentials`, part of me thinks this function should live in main.go but the other half isn't so sure

Reply:
This was the last thing I added, it used to live in e3db-go but wanted to move it to this repo. I am happy to put it in main. Main felt very network focused and while this is one layer up it still seemed more crypto than network 🤷🏻‍♂️